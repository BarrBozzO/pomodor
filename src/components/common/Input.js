import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const StyledInput = styled.input`
  border: 1px solid #ddd;
  font-size: 14px;
  padding: 11px;
  color: #333;
  border-radius: 4px;
`;

function Input(props) {
  const {
    name,
    disabled,
    value,
    type,
    placeholder,
    handleChange,
    className,
    step
  } = props;
  return (
    <StyledInput
      name={name}
      type={type}
      disabled={disabled}
      value={value}
      placeholder={placeholder}
      onChange={handleChange}
      className={className}
      step={step}
    />
  );
}

Input.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  disabled: PropTypes.bool,
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  className: PropTypes.string,
  step: PropTypes.number
};

Input.defaultProps = {
  disabled: false,
  primary: false,
  placeholder: "",
  type: "text" // text or number
};

export default Input;
