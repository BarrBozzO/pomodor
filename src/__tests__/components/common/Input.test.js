import React from "react";
import renderer from "react-test-renderer";

import Input from "../../../components/common/Input";

it("should render Input component correctly when all props are passed", () => {
  const name = "testInput",
    type = "password",
    disabled = true,
    value = "test",
    placeholder = "test placeholder",
    onChange = jest.fn();

  const tree = renderer
    .create(
      <Input
        name={name}
        type={type}
        disabled={disabled}
        value={value}
        placeholder={placeholder}
        handleChange={onChange}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
