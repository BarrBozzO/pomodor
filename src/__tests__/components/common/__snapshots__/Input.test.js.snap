// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Input component correctly when all props are passed 1`] = `
<input
  className="sc-bdVaJa jOMGgo"
  disabled={true}
  name="testInput"
  onChange={[MockFunction]}
  placeholder="test placeholder"
  type="password"
  value="test"
/>
`;
